pkg/pmem-device-manager/pmd-lvm.go

```diff
@@ -343,7 +343,9 @@ func createNS(r *ndctl.Region, percentage uint) error {
 			Size:  canUse,
 			Align: align,
 		})
-		return fmt.Errorf("failed to create PMEM namespace with size '%d' in region '%s': %v", canUse, r.DeviceName(), err)
+		if err != nil {
```

Comment:
I am surprised how this state of code ever worked, as it always returns error.... I am quite sure I wrote that if stmt there when this code was developed. But seems between commits 5365a347fb2417b74705fe093579e686134b6140 and 0748a9125e49556404c9d41449c2c87ddb36edf7 that whole function was moved, but leaving those (quite important ) if-lines behind. Just curious, what was reasoning to leave out if-part. And how the testing actually passed then. We dont have tests for that part?

Reply:
means, current fix is not new code, but actually restores code that was there originally.

Reply:
You are right @okartau, It was unintentionally introduced by me while moving the code as part of removing init-container from the driver.

Reply:
> And how the testing actually passed then. We dont have tests for that part?

We do not have tests to cover this part of the code. Current tests cover only create, delete, and list devices on a pre-populated logical volume group.

Reply:
We *do* go through this code once when installing a driver in LVM mode for the first time. The error return then causes a restart of the container. The next instance of the container then gets passed this error because it uses the then existing namespace instead of entering this faulty branch again.

The new check for "no restarts" failed when this line wasn't fixed and passed once it was.